compiler-rt/lib/asan/asan_shadow_setup.cpp

```diff
@@ -91,9 +91,25 @@ void InitializeShadowMemory() {
       ReserveShadowMemoryRange(shadow_start, kLowShadowEnd, "low shadow");
     // mmap the high shadow.
     ReserveShadowMemoryRange(kHighShadowBeg, kHighShadowEnd, "high shadow");
-    // protect the gap.
-    ProtectGap(kShadowGapBeg, kShadowGapEnd - kShadowGapBeg + 1);
-    CHECK_EQ(kShadowGapEnd, kHighShadowBeg - 1);
+#  if SANITIZER_AIX == 1 && SANITIZER_WORDSIZE == 64
```

Comment:
all this nesting is hard to read
maybe we need a refactoring to make it more generic